### ISSUE TITLE
[BUG] Removing the default storageclass 'longhorn' works for ~15-30 seconds, but it's reappears

### ISSUE THREAD

**Issue by ws-way** [longhorn/longhorn#2119]
With v1.0.2 we did remove the default installed 'longhorn' storageclass, created our own storageclasses and made 1 of those own created storageclasses the default one.

With v1.1.0 we can remove the default installed 'longhorn' storageclass, but it reappears after ~15-30 seconds, resulting in that there are 2 default storageclasses.

```
kubectl delete storageclass longhorn
storageclass.storage.k8s.io "longhorn" deleted
```
After ~15 - 30 seconds it reappears and we have 2 defaults:
```
kubectl get storageclasses
NAME                                PROVISIONER          RECLAIMPOLICY   VOLUMEBINDINGMODE   ALLOWVOLUMEEXPANSION   AGE
longhorn (default)                  driver.longhorn.io   Delete          Immediate           true                   4m16s
longhorn-fast-1-replica (default)   driver.longhorn.io   Retain          Immediate           true                   5m52s
longhorn-fast-2-replica             driver.longhorn.io   Retain          Immediate           true                   5m52s
longhorn-fast-3-replica             driver.longhorn.io   Retain          Immediate           true                   5m52s
longhorn-slow-1-replica             driver.longhorn.io   Retain          Immediate           true                   5m52s
longhorn-slow-2-replica             driver.longhorn.io   Retain          Immediate           true                   5m52s
longhorn-slow-3-replica             driver.longhorn.io   Retain          Immediate           true
```

**Comment by PhanLe1010** [comment]
@ws-way 

In Longhorn v1.1.0, users cannot directly update `longhorn` StorageClass. In order to update `longhorn` StorageClass, you have to update `longhorn` StorageClass ConfigMap in `longhorn-system` namespace:
```
kubectl edit configmap longhorn-storageclass -n longhorn-system
```
Because we have [some issues](https://github.com/longhorn/longhorn/issues/1527) with StorageClass that prevent users from upgrading Longhorn using Helm if they modified Longhorn StorageCLass, we changed the logic for handling StorageClass modification. More specifically, if users want to modify `longhorn` storageclass, users have to modify the `longhorn` StorageClass ConfigMap in `longhorn-system` namespace

**Comment by ws-way** [comment]
Thanks, I already saw it on https://github.com/longhorn/longhorn-manager/pull/735 . Is there any way to prevent the creation of the default storage class with the helm install?

**Comment by PhanLe1010** [comment]
Yeah, you can `-set persistence.defaultClass=false` when doing Helm installation.

**Comment by ws-way** [comment]
I already tested that, but it doesn't seem to work. I will try it again, maybe I mixed up something

**Comment by PhanLe1010** [comment]
@ws-way 
Sorry. I might get your questions wrong. 

Updated the response:

You can `--set persistence.defaultClass=false` when doing Helm installation to indicate that you don't want to set `longhorn` as the default StorageClass.

`longhorn` StorageClass is still created though. There isn't away to prevent the creation of `longhorn` StorageClass when installing by Helm.
